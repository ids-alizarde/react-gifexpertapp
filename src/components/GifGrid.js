import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../services/GetGifs';

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    // useEffect( () => {
    //     getGifs( category )
    //         .then( setImages );
    // }, [ category ]);

    const { data:images, loading } = useFetchGifs( category );

    return (
        <>
            <h3 className="animate__animated animate__fadeInTopLeft">{ category }</h3>

            { loading && <p>Loading.........</p> }

            <div className="card-grid">

                { images.map( img => (

                    <GifGridItem 
                        key={img.id}
                        { ...img }>

                    </GifGridItem>

                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
