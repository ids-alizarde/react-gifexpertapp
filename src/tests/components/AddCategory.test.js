import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en el componente <AddCategory />', () => {

    // const setCategories = () => {};
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories = { setCategories } /> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories = { setCategories } /> );
    })
    
    test('Debe mostrar correctamente el componenete <AddCategory />', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe de cambiar el contenido del input', () => {

        const input = wrapper.find( 'input' );
        const value = 'Paw patrol';

        input.simulate( 'change', { target: { value } });

    });

    test('NO debe de mandar la informacion con submit', () => {

        wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} });
        expect( setCategories ).not.toHaveBeenCalled();

    });

    test('Debe de llamar el setCategories y limpiar el input', () => {
        
        let value = 'Mazinger Z';
        wrapper.find( 'input' ).simulate( 'change', { target: { value } });
        wrapper.find( 'form' ).simulate( 'submit', { preventDefault(){} });
        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );

        value = '';
        // wrapper.find( 'input' ).simulate( 'change', { target: { value } });
        // expect( wrapper.find( 'input' ).text() ).toBe( value );

        expect( wrapper.find( 'input' ).props().value ).toBe ( value );

    });
    
});
