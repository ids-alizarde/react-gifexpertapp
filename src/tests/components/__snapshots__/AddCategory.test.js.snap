// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <AddCategory /> Debe mostrar correctamente el componenete <AddCategory /> 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
